packages/jellyfish-api-core/__tests__/category/account/getPendingFutureSwaps.test.ts

```diff
@@ -0,0 +1,598 @@
+        expect(pendingFutureSwaps2).toStrictEqual(pendingFutureSwaps1) // Nothing change
+
+        const idTSLA = await testing.token.getTokenId('TSLA')
+        await testing.rpc.masternode.setGov({ ATTRIBUTES: { [`v0/token/${idTSLA}/dfip2203`]: 'false' } })
+        await testing.generate(1)
+
+        const attributes = await testing.rpc.masternode.getGov('ATTRIBUTES')
+        expect(attributes.ATTRIBUTES[`v0/token/${idTSLA}/dfip2203`]).toStrictEqual('false')
+
+        // Call getpendingfutureswaps after active is set to false
+        {
+          const pendingFutureSwaps = await testing.rpc.account.getPendingFutureSwaps(tslaAddress)
+          expect(pendingFutureSwaps).toStrictEqual({
+            owner: tslaAddress,
+            values: []
+          })
+        }
+      })
+
+      it('Should getPendingFutureSwaps if futureswap DUSD for TSLA', async () => {
+        // Call getpendingfutureswaps before performing futureswap
+        {
+          const pendingFutureSwaps = await testing.rpc.account.getPendingFutureSwaps(tslaAddress)
+          expect(pendingFutureSwaps).toStrictEqual({
+            owner: tslaAddress,
+            values: []
+          })
+        }
+
+        // futureswap
+        await testing.container.call('futureswap', [tslaAddress, '1@DUSD', 'TSLA'])
+        await testing.generate(1)
+
+        // Call getpendingfutureswaps after performing futureswap
+        const pendingFutureSwaps1 = await testing.rpc.account.getPendingFutureSwaps(tslaAddress)
+        expect(pendingFutureSwaps1).toStrictEqual({
+          owner: tslaAddress,
+          values: [
+            {
+              source: '1.00000000@DUSD',
+              destination: 'TSLA'
+            }
+          ]
+        })
+
+        // Wait for 1 block before the next settle block
+        const nextSettleBlock = await testing.container.call('getfutureswapblock')
+        await testing.generate(nextSettleBlock - await testing.rpc.blockchain.getBlockCount() - 1)
+
+        const pendingFutureSwaps2 = await testing.rpc.account.getPendingFutureSwaps(tslaAddress)
+        expect(pendingFutureSwaps2).toStrictEqual(pendingFutureSwaps1) // Nothing change
+
+        const idTSLA = await testing.token.getTokenId('TSLA')
+        await testing.rpc.masternode.setGov({ ATTRIBUTES: { [`v0/token/${idTSLA}/dfip2203`]: 'false' } })
+        await testing.generate(1)
+
+        const attributes = await testing.rpc.masternode.getGov('ATTRIBUTES')
+        expect(attributes.ATTRIBUTES[`v0/token/${idTSLA}/dfip2203`]).toStrictEqual('false')
+
+        // Call getpendingfutureswaps after active is set to false
+        {
+          const pendingFutureSwaps = await testing.rpc.account.getPendingFutureSwaps(tslaAddress)
+          expect(pendingFutureSwaps).toStrictEqual({
+            owner: tslaAddress,
+            values: []
+          })
+        }
+      })
+    })
+  })
+
+  describe('Multiple futureswaps', () => {
```

Comment:
I don't duplicate the negative test units for setGov in **multiple** futureswaps